kubernetes.go

```diff
@@ -78,11 +80,45 @@ func NewInClusterK8sClient() (K8sClient, error) {
 	}}
 	httpClient := &http.Client{Transport: transport, Timeout: time.Nanosecond * 0}
 
-	return &k8sClient{
+	client := &k8sClient{
 		host:       "https://" + net.JoinHostPort(host, port),
 		token:      string(token),
 		httpClient: httpClient,
-	}, nil
+	}
+
+	// Create a new file watcher to listen for new Service Account tokens
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+
+	go func() {
+		for {
+			select {
+			case event, ok := <-watcher.Events:
+				if !ok {
+					return
+				}
+				if event.Op&fsnotify.Write == fsnotify.Write {
+					token, err := ioutil.ReadFile(serviceAccountToken)
+					if err == nil {
+						client.token = string(token)
```

Comment:
Analising this change, Some problems a race condition could happen. I think this update should be lock and unlock by RWMutex